CMakeLists.txt

```diff
@@ -163,6 +163,7 @@ endif()
 
 cmake_push_check_state()
 
+set(CMAKE_REQUIRED_FLAGS -std=c++11)
```

Comment:
Why is this necessary if `add_compile_options(-std=c++11)` exists in [CMakeLists.txt:24](https://github.com/ledger/ledger/blob/master/CMakeLists.txt#L24)?

Reply:
I was compiling on a machine with ICU 75.1, which requires at least C++ 17.

https://github.com/ledger/ledger/blob/0c323a569f2774431fcef2e5fe4b5c425ab572b2/CMakeLists.txt#L167-L168

So while `check_cxx_source_runs` for `boost/regex/icu.hpp` (which includes headers from ICU) passes when C++ version is not specified, if Ledger (`-std=c++11`) includes `boost/regex/icu.hpp`, the build fails.

https://github.com/ledger/ledger/blob/0c323a569f2774431fcef2e5fe4b5c425ab572b2/CMakeLists.txt#L24

It is not directly related to Emscripten support. Should I revert it?

Reply:
Thanks for the added context, @gudzpoz, very helpful.

Now I see… before line 166 there is `cmake_push_check_state()`, so setting the flag here is for only for the `check_cxx_source_runs` "context".

Reading the doucmentation for [`check_cxx_source_runs`](https://cmake.org/cmake/help/latest/module/CheckCXXSourceRuns.html) it offers a `CMAKE_REQUIRED_FLAGS` parameter that is 
> String of additional flags to pass to the compiler

Maybe that's helpful?